apps/src/turtle/SaveImageButton.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import msg from '@cdo/locale';
+import * as imageUtils from '@cdo/apps/imageUtils';
+import color from '@cdo/apps/util/color';
+
+const styles = {
+  icon: {
+    lineHeight: 'inherit',
+    color: color.white,
+    paddingRight: 8
+  },
+  container: {
+    lineHeight: '40px',
+    textAlign: 'center',
+    verticalAlign: 'middle',
+    display: 'inline-block',
+    textSize: 'large'
+  },
+  button: {
+    padding: '0px 8px',
+    lineHeight: '40px',
+    borderRadius: 5,
+    backgroundColor: color.cyan,
+    borderColor: color.cyan,
+    color: color.white
+  }
+};
+
+class SaveImageButton extends React.Component {
+  save = async () => {
+    const canvasEl = document.getElementById('display');
+    const blob = await imageUtils.canvasToBlob(canvasEl);
+    const download = document.createElement('a');
+    download.href = URL.createObjectURL(blob);
+    download.download = 'image.png';
+    download.click();
+  };
```

Comment:
questions/suggestion:
1. this seems like useful shared code -- should we move it to `imageUtils`?
2. 'display' should also be a required prop for this component (something like `targetElementId` or `targetImageId`). we could alternatively pass the element as a prop instead of the id
3. where does the 'display' id come from? i'm having trouble finding it in `<ArtistVisualizationColumn/>`

Reply:
1. Done!
2. Done!
3. It is created here: https://github.com/code-dot-org/artist/blob/main/visualization.js#L106 and then added to the DOM here: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/turtle/artist.js#L507